### PR TITLE
Bug 1849833 - Fix warnings and deprecations in build files.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1849833

### CODE REVIEW

File: android-components/components/browser/session-storage/build.gradle [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (gabrielluong) [inline android-components/components/browser/session-storage/build.gradle:1]: Could we move this below the license?

File: android-components/components/browser/state/build.gradle [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (gabrielluong) [inline android-components/components/browser/state/build.gradle:1]: Same as above

File: android-components/components/browser/thumbnails/build.gradle [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (gabrielluong) [inline android-components/components/browser/thumbnails/build.gradle:1]: Same as above

File: android-components/components/compose/awesomebar/build.gradle [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (gabrielluong) [inline android-components/components/compose/awesomebar/build.gradle:1]: Same as above

File: android-components/components/feature/accounts-push/build.gradle [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (gabrielluong) [inline android-components/components/feature/accounts-push/build.gradle:1]: Same as above, but not gonna point each one anymore. Let's just make sure we go through all the files.

**Review by gabrielluong — APPROVED** [review]

**Comment by mcarare** [comment]
landing this on Monday.

**Pull request merged**